packages/api/portfolios/environments/createEnvironment.ts

```diff
@@ -0,0 +1,143 @@
+import "reflect-metadata";
+import { createConnection } from "../../utils/database";
+import { Portfolio } from "../../../orm/entity/Portfolio";
+import { Application } from "../../../orm/entity/Application";
+import { Environment, IEnvironmentCreate } from "../../../orm/entity/Environment";
+import { EnvironmentRepository } from "../../repository/EnvironmentRepository";
+import { APIGatewayProxyEventPathParameters, APIGatewayProxyResult, Context } from "aws-lambda";
+import { DATABASE_ERROR, DUPLICATE_ENVIRONMENT_NAME, NO_SUCH_PORTFOLIO_OR_APPLICATION } from "../../utils/errors";
+import { ApiSuccessResponse, SuccessStatusCode } from "../../utils/response";
+import internalSchema = require("../../models/internalSchema.json");
+import middy from "@middy/core";
+import xssSanitizer from "../../portfolioDrafts/xssSanitizer";
+import jsonBodyParser from "@middy/http-json-body-parser";
+import validator from "@middy/validator";
+import JSONErrorHandlerMiddleware from "middy-middleware-json-error-handler";
+import cors from "@middy/http-cors";
+import { ApiGatewayEventParsed } from "../../utils/eventHandlingTool";
+import { CORS_CONFIGURATION } from "../../utils/corsConfig";
+import { wrapSchema } from "../../utils/schemaWrapper";
+import { errorHandlingMiddleware } from "../../utils/errorHandlingMiddleware";
+import { isValidUuidV4 } from "../../utils/validation";
+import createError from "http-errors";
+import { IpCheckerMiddleware } from "../../utils/ipLogging";
+
+/**
+ * Submits the environment of an application
+ *
+ * @param event - The POST request from API Gateway
+ */
+export async function baseHandler(
+  event: ApiGatewayEventParsed<IEnvironmentCreate>,
+  context?: Context
+): Promise<APIGatewayProxyResult> {
+  const parameters = Object.entries(event.pathParameters as APIGatewayProxyEventPathParameters);
+  if (
+    !parameters.every(
+      ([param, paramValue]) => ["portfolioId", "applicationId"].includes(param) && isValidUuidV4(paramValue)
+    )
+  ) {
+    throw createError(404, "Shape validation failed, invalid UUIDv4");
+  }
```

Comment:
Do we have an error we can return for this?

Reply:
Or is this just here to throw an error to the middleware

Reply:
This was intended to throw the error to the middleware. It's similar to the `validateRequestShape` function.

Reply:
Updated to remove the `catch` block and all errors handled by `errorHandlingMiddleware`. it's not pretty, but it works.